client/src/session_state.rs

```diff
@@ -49,6 +48,10 @@ lazy_static! {
 pub(crate) struct SessionState {
     /// A unique identifier for the session, this is NOT the session id assigned after a session is created
     id: u32,
+    /// Time offset between the client and the server.
+    client_offset: Duration,
```

Comment:
Obviously there is still potential for time to drift during the session

Reply:
True... Yet it updates the offset every time the secure channel is (re)created or renewed.